src/System.IO.Pipelines/System/IO/Pipelines/ReadableBuffer.cs

```diff
@@ -201,18 +221,50 @@ public BufferEnumerator GetEnumerator()
             return new BufferEnumerator(BufferStart, BufferEnd);
         }
 
+        public ReadCursor Move(ReadCursor cursor, long count)
```

Comment:
This method is strange. It's not clear if it offsets or moves to a new location. Also, it's strange that it moves cursor but it's an instance method on RB.

Reply:
It needs `End` cursor to check bounds, that's why it's on RB.